### PR TITLE
[PWA, Service Worker] Migrate to PWA and cache files using Service Worker

### PR DESCRIPTION
**Pull request by aleksazh**
Related to issues #15 and #16.

Convert app to PWA:

* [Migrate your site to a Progressive Web App](https://codelabs.developers.google.com/codelabs/migrate-to-progressive-web-apps/#0)
* [The Web App Manifest](https://developers.google.com/web/fundamentals/web-app-manifest/)

Cache files using service worker for offline mode:

* [Caching Files with Service Worker](https://developers.google.com/web/ilt/pwa/caching-files-with-service-worker)

Generate service worker with Workbox CLI:

* [Generate a Complete Service Worker with Workbox CLI](https://developers.google.com/web/tools/workbox/guides/generate-service-worker/cli)

Next improvements:
1. Delete outdated cache
2. Fullscreen mode?
3. Orientation?
4.

### CODE REVIEW

File: sw.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+});
+
+/**
+ * The workboxSW.precacheAndRoute() method efficiently caches and responds to
+ * requests for URLs in the manifest.
+ * See https://goo.gl/S9QRab
+ */
+self.__precacheManifest = [
+  {
+    "url": "build/asm.js/opencv_js.js",
```
Comment (riju) [inline sw.js]: Now we have deleted the build/asm.js

File: sw.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    "revision": "f4e04cfb990e8eafb1fb2909f05e4233"
+  },
+  {
+    "url": "build/asm.js/opencv.js",
+    "revision": "0040389865e99966926204e15255b006"
+  },
+  {
+    "url": "build/opencv.js",
```
Comment (riju) [inline sw.js]: and also this

File: sw.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    "revision": "34d483272ed536affc5cfb225497b991"
+  },
+  {
+    "url": "build/wasm/opencv_js.js",
+    "revision": "1b73209d47ad04f258486f749c41e601"
+  },
+  {
+    "url": "build/wasm/opencv.js",
```
Comment (riju) [inline sw.js]: we might want to cache the opencv_js.worker.js also

File: sw.js (hunk 1 above)
Comment (aleksazh) [reply sw.js]: Yes, I updated sw.js automatically with workbox and forgot to delete asm files after that.

**Pull request merged**